### ISSUE TITLE
Stack invocations are platform dependent 

### ISSUE THREAD

**Issue by domenkozar** [input-output-hk/stack2nix#56]
If you have in your cabal file:

 ```
if os(linux)
   build-depends: systemd
```

Then invoking `stack list-dependencies` on macOS won't return systemd, although it's needed for linux.

Ideally we'd deal here the same way as for cabal2nix, specifying what platform we're asking dependencies for.

Bug was discovered in https://github.com/input-output-hk/cardano-sl/pull/1293

Use the same hardcoded value for now as in https://github.com/input-output-hk/stack2nix/pull/54

**Comment by domenkozar** [comment]
Best to use `platformL` on `Config` to hardcode it as part of #55

**Comment by domenkozar** [comment]
In e77118f989998ab6ec57958d4cc70d41eca71c32 I managed to hardcode the platform to darwin within `RIO` monad stack and run stack2nix on linux.

I would expect that running stack2nix on cardano-sl wouldn't include `systemd` in stack2nix plan output.

To reproduce:

```
git clone https://github.com/input-output-hk/stack2nix.git
cd stack2nix
stack install --nix
git clone https://github.com/input-output-hk/cardano-sl.git
cd cardano-sl
~/.local/bin/stack2nix . > pkgs/default.nix
```

All stack planning happens in `stack2nix/src/Stack2nix/External/Stack.hs`

**Comment by domenkozar** [comment]
The culprit was that `systemd` is not built only for windows, probably to avoid this bug in the first place.

**Comment by domenkozar** [comment]
Fixed in #79
